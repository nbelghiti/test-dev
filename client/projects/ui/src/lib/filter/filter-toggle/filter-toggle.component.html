<div class="c-filter-toggle" [ngClass]="{
	'c-filter-toggle--inline': inline,
	'c-filter-toggle--borderless': borderless,
	'c-filter-toggle--compact': compact,
	'c-filter-toggle--transparent': background === 'transparent',
	'c-filter-toggle--grey': background === 'grey',
	'c-filter-toggle--large': large
}">
	<span *ngIf="label" class="c-filter-toggle__label">{{label}}</span>

	<button class="c-filter-toggle__trigger" type="button">
		<span class="c-filter-toggle__current-option">{{currentOption}}</span>

		<div class="c-filter-toggle__icon">
			<ui-icon name="caret-down" size="small"></ui-icon>
		</div>
	</button>
</div>
