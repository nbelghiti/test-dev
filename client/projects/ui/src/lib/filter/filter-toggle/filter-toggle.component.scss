// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: User Info
// =============================================================================
.c-filter-toggle {
	display: flex;
	flex-direction: column;

	&:hover {

		.c-filter-toggle__trigger {
			border: $border-width-base solid $border-color-black;
		}
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-filter-toggle__label {
	margin-bottom: 0.2rem;

	font-size: $font-size-2x-small;
	line-height: $line-height-base;

	color: $text-color-grey-46;
}

.c-filter-toggle__trigger {
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 0.6rem 2rem;

	white-space: nowrap;

	background: none;
	border: $border-width-base solid $border-color-base;
	background-color: $background-color-base;

	transition:
		border $animation-base,
		background-color $animation-base,
		box-shadow $animation-base;

	cursor: pointer;

	&:focus {
		border: $border-width-base solid $border-color-black;
	}

	&:hover {

		.c-filter-toggle__current-option {
			text-decoration: underline;
		}
	}
}

.c-filter-toggle__current-option {
	overflow: hidden;
	margin-right: 1rem;

	font-weight: $font-weight-ultrabold;
	font-size: $font-size-2x-small;
	text-overflow: ellipsis;
	white-space: nowrap;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-filter-toggle--inline {
	flex-direction: row;
	align-items: center;

	.c-filter-toggle__label {
		margin-right: 1rem;
		margin-bottom: 0;

		color: $text-color-base;
	}

	.c-filter-toggle__trigger {
		flex-grow: 1;
	}
}

.c-filter-toggle--borderless {

	.c-filter-toggle__trigger {
		border-color: transparent;
	}

	&:hover {

		.c-filter-toggle__trigger {
			border-color: transparent;
			background-color: $background-color-grey-96;
		}
	}
}

.c-filter-toggle--compact {

	.c-filter-toggle__trigger {
		padding: 0 1rem;
	}
}

.c-filter-toggle--transparent {

	.c-filter-toggle__trigger {
		background-color: transparent;
	}
}

.c-filter-toggle--grey {

	.c-filter-toggle__trigger {
		color: $text-color-grey-46;
		background-color: $background-color-grey-96;

		&:focus {
			color: $text-color-base;
		}
	}

	&.c-filter-toggle--borderless:hover {

		.c-filter-toggle__trigger {
			box-shadow: 0 0 0 $border-width-base $border-color-black;
		}
	}
}

.c-filter-toggle--large {

	.c-filter-toggle__current-option {
		font-size: $font-size-1x-large;

		&:focus {
			color: $text-color-base;
		}
	}

	.c-filter-toggle__trigger {
		padding: 0;

		border: none;

		&:hover {
			border: none;
		}
	}
}
