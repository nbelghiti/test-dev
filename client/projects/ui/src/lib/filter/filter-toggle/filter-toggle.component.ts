import { Component, Input } from '@angular/core';

import { FilterToggleBackground } from './filter-toggle.component.types';

@Component({
	selector: 'ui-filter-toggle',
	templateUrl: './filter-toggle.component.html',
	styleUrls: ['./filter-toggle.component.scss']
})

export class FilterToggleComponent {
	@Input() label: string;
	@Input() currentOption: string = '- All';
	@Input() inline?: boolean = false;
	@Input() borderless?: boolean = false;
	@Input() compact?: boolean = false;
	@Input() large?: boolean = false;
	@Input() background?: FilterToggleBackground = FilterToggleBackground.default;
}
