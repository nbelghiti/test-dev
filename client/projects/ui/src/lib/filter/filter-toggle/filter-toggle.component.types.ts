export enum FilterToggleBackground {
	default = '',
	transparent = 'transparent',
	grey = 'grey',
}
