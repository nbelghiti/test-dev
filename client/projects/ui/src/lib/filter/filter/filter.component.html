<div uiFlyout class="c-filter">
	<div uiFlyoutToggle>
		<ng-content select="ui-filter-toggle"></ng-content>
	</div>

	<div uiFlyoutContent class="c-filter__content c-flyout__content--left {{classes}}" uiFlyoutToggle [toggleState]="FlyoutState.CLOSED">
		<ng-content select="ui-filter-menu"></ng-content>
	</div>
</div>
