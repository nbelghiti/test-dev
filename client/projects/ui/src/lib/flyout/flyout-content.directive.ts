import { Directive, HostBinding, Input } from '@angular/core';

@Directive({
	selector: '[uiFlyoutContent]'
})
export class FlyoutContentDirective {
	@Input() public class: string;

	@HostBinding('class') public get classListIfEmpty(): string {
		return `${this.class} c-flyout__content`;
	}
}
