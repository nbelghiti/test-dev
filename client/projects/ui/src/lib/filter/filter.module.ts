import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { FlyoutModule } from '../flyout/flyout.module';
import { IconModule } from '../icon/icon.module';

import { FilterMenuItemComponent } from './filter-menu-item/filter-menu-item.component';
import { FilterMenuComponent } from './filter-menu/filter-menu.component';
import { FilterToggleComponent } from './filter-toggle/filter-toggle.component';
import { FilterComponent } from './filter/filter.component';

@NgModule({
	imports: [
		CommonModule,
		RouterModule,
		IconModule,
		FlyoutModule,
	],
	declarations: [
		FilterComponent,
		FilterMenuComponent,
		FilterMenuItemComponent,
		FilterToggleComponent,
	],
	exports: [
		FilterComponent,
		FilterMenuComponent,
		FilterMenuItemComponent,
		FilterToggleComponent,
	],
})
export class FilterModule {}
