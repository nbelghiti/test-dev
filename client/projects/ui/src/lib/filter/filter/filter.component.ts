import { Component, Input } from '@angular/core';

import { FlyoutState } from '../../flyout/flyout.types';

@Component({
	selector: 'ui-filter',
	templateUrl: './filter.component.html',
})

export class FilterComponent {
	@Input() public classes: string;

	public FlyoutState = FlyoutState;
}
