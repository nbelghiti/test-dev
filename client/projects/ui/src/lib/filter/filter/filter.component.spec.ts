import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { FlyoutModule } from '../../flyout/flyout.module';
import { IconComponent } from '../../icon/icon.component';
import { FilterToggleComponent } from '../filter-toggle/filter-toggle.component';

import { FilterComponent } from './filter.component';

describe('FilterComponent', () => {
	let component: FilterComponent;
	let fixture: ComponentFixture<FilterComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				FlyoutModule,
			],
			declarations: [
				FilterComponent,
				FilterToggleComponent,
				IconComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(FilterComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
