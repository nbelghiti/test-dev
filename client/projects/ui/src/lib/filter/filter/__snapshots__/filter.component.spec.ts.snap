// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComponent Should render correctly 1`] = `
<ui-filter
  FlyoutState={[Function Object]}
>
  <div
    class="c-filter c-flyout is-closed"
    uiflyout=""
  >
    <div
      class="c-flyout__toggle"
      uiflyouttoggle=""
    />
    <div
      class="c-filter__content c-flyout__content--left c-flyout__content c-flyout__toggle"
      ng-reflect-class="c-filter__content c-flyout__co"
      ng-reflect-toggle-state="CLOSED"
      uiflyoutcontent=""
      uiflyouttoggle=""
    />
  </div>
</ui-filter>
`;
